frontend/src/backend/files.rs

```diff
@@ -129,13 +130,39 @@ pub async fn get_directory(id: Id) -> Result<FileDirectory, String> {
             "get_directory".to_string(),
             Some(&serde_json::json!({ "id": id })),
         )
-        .await;
+            .await;
     } else {
         // user is offline throw a error
         return Err("user is offline".to_string());
     }
 }
 
+pub async fn get_file(id: Id) -> Result<Option<FileNode>, String> {
```

Comment:
we need to handle the different situations here
Whether the file already exist in your dogs are we should fetch it again from the IB Bsckend, or even whether we should fetch it from the searialdb later